src/Util/JsonLD.php

```diff
@@ -230,15 +228,15 @@ public static function fetchElement($array, $element, $key = '@id', $type = null
 
 		if (is_null($type) || is_null($type_value)) {
 			$element_array = self::fetchElementArray($array, $element, $key);
-			if (is_null($element_array)) {
+			if (empty($element_array)) {
```

Comment:
The problem with checking via `empty(...)` is that we wouldn't knew the difference between no value and a deliberately empty array.

Reply:
In this context, it doesn't matter since the following code expects an array with at least one element (for the `array_shift` or the `foreach`).

Reply:
This is a really fragile part of the code. I have to have a deeper look at it and at all consequences, but currently it is too early for me.

Reply:
I stand by this choice to ensure return type is not null. I understand the difference between `no element` and `empty array element` but I looked at all the 6 calls of `JsonLD::fetchElementArray` and the outcome wouldn't be different in either case.

Reply:
This weekend I will try to create test cases for the parser part. I guess it is a good idea to test that part of the code - and hopefully these tests will not be too hard to code.

Reply:
For JsonLD or ActivityPub\Receiver or Processor?

Reply:
For the JsonLD parser. This thing is crucial for the communication. And testing this should be easy. It's just providing some array, then fetching data, then comparing this with the expected value.

Reply:
Absolutely. Is there any standard document reference?

Reply:
Well, only you know what you wanted to do with this parser, so I'll leave you to it. I'll rebase my branch onto your tests, but I'm guessing you will run into the same oddity than I did with the default value for the `$key` parameter.

Reply:
I will have a try today.

Reply:
I added the first tests. Since this seemed to work, I will now add some more tests to cover all situations for the two parser functions.